internal/tracing/init.go

```diff
@@ -124,5 +163,31 @@ func OpenTelemetryInit() error {
 		panic(fmt.Sprintf("OpenTelemetry error: %v", err))
 	}))
 
-	return nil
+	return ctx, nil
+}
+
+// envMapCarrier is a simple implementation of the TextMapCarrier interface
+// that uses a map to store the key-value pairs.
+// This is used to carry the extracted traceparent and tracestate
+// from the environment variables into the context.
+type envMapCarrier struct {
```

Comment:
Is this different in some way from [`propagation.MapCarrier`](https://pkg.go.dev/go.opentelemetry.io/otel@v1.35.0/propagation#MapCarrier)? If so, I think it would be helpful to describe the important differences in a comment so future maintainers are less tempted to swap these. 😀